independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/api/WebClientApplicationException.java

```diff
@@ -24,35 +25,36 @@
 @SuppressWarnings("serial")
 public class WebClientApplicationException extends WebApplicationException {
 
-    public WebClientApplicationException() {
-        super();
+    public WebClientApplicationException(int responseStatus) {
+        this(responseStatus, null);
     }
 
-    public WebClientApplicationException(String reason) {
-        super(reason, null, DummyResponse.INSTANCE);
+    public WebClientApplicationException(int responseStatus, String responseReasonPhrase) {
+        super("Server response is: " + responseStatus, null, new DummyResponse(responseStatus, responseReasonPhrase));
     }
 
-    @Override
-    public final Response getResponse() {
-        return null;
-    }
-
-    // used in order to avoid depending on the server parts just for generating the exception
+    /**
+     * Used in order to avoid depending on the server parts just for generating the exception
+     * The
```

Comment:
Premature EOL? 😉

Reply:
It doesn't surprise me one bit that I did this :)